pkg/imgpkg/bundle/images_refs.go

```diff
@@ -17,11 +17,42 @@ import (
 
 type ImageRef struct {
 	lockconfig.ImageRef
-	IsBundle *bool
+	IsBundle  *bool
+	Copiable  bool
+	ImageType ImageType
+}
+
+// ImageType defines the type of Image
+type ImageType string
+
+const (
+	// BundleImage Image that represents a Bundle
+	BundleImage ImageType = "Bundle"
+	// ContentImage Image that is part of the Bundle
+	ContentImage ImageType = "Image"
+	// SignatureImage Image that contains a signature
+	SignatureImage ImageType = "Signature"
+	// InternalImage Image that contains a signature
+	InternalImage ImageType = "Internal"
+)
+
+// NewContentImageRef Constructs a new ImageRef for Bundle or Image
+func NewContentImageRef(imgRef lockconfig.ImageRef, isBundle bool) ImageRef {
```

Comment:
"naked" bool as an arg reads very confusing at the callee. i would recommend making this into:

~NewContentImageRef~
+NewBundleImageRef
+NewContentImageRef

(even in the enum you are making a distinction between ContentImage and BundleImage).

Reply:
or just always use NewImageRefWithType, but remove bool from signature, so that isBundle is derived based on type.

Reply:
done